aea/protocols/generator/extract_specification.py

```diff
@@ -0,0 +1,262 @@
+# -*- coding: utf-8 -*-
```

Comment:
This module now has the logic to extract protocol specification fields into pythonic equivalences. 

It has one function `extract()` with all its private sub-functions. It extracts all specification fields into python objects and encapsulates them into a `PythonicProtocolSpecification` class.

I separated these code into its own module mainly to declutter the main generator and because they have a nice logical grouping.